dspace-server-webapp/src/main/java/org/dspace/app/rest/security/OidcAuthenticationFilter.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.security;
+
+import static org.dspace.authenticate.OidcAuthenticationBean.OIDC_AUTH_ATTRIBUTE;
+
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+
+/**
+ * This class will filter openID Connect requests and try and authenticate them.
+ *
+ * @author Pasquale Cavallo (pasquale.cavallo at 4science dot it)
+ */
+
+public class OidcAuthenticationFilter extends StatelessLoginFilter {
```

Comment:
This should be renamed `OidcLoginFilter` as it extends `StatelessLoginFilter` (and not `StatelessAuthenticationFilter`, which is a different class).  We don't want to imply this extends `StatelessAuthenticationFilter`.  Again, we had to fix this for Shibboleth as well in #8032